### PR TITLE
feat: [IDP-576]: Support for editing harness ci-cd plugin behind FF.

### PR DESCRIPTION
**Pull request by devesh-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48352)

### CODE REVIEW

**Comment by devesh-harness** [comment]
trigger uts

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java [hunk 1]
```diff
@@ -763,7 +763,8 @@ public enum FeatureName {
       "Enabling support for HPA and PDB kind resources in k8s deployments in NG", HarnessTeam.CDP),
   SRM_ENABLE_BASELINE_BASED_VERIFICATION("This is used to enable baseline based verification.", HarnessTeam.CV),
   SPG_DISABLE_CUSTOM_WEBHOOK_V3_URL("This is used to disable customer webhook authentication.", SPG),
-  SRM_MICRO_FRONTEND("This FF is used to enable the micro-frontend for SRM", CV);
+  SRM_MICRO_FRONTEND("This FF is used to enable the micro-frontend for SRM", CV),
+  IDP_ENABLE_EDIT_HARNESS_CI_CD_PLUGIN("This FF is for allowing user to edit config for harness-ci-cd plugin from UI", HarnessTeam.IDP);
```
Comment (vigneswara-propelo) [inline 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:767]: How do we check for this flag for editing? Will UI make separate call for this?

File: idp-service/src/main/resources/metadata/harness-ci-cd.yaml [hunk 2]
```diff
@@ -9,10 +9,10 @@ iconUrl: https://static.harness.io/ng-static/images/favicon.png
 imageUrl: ''
 config: |-
   proxy:
-    '/harness':
+    '/harness/prod':
       target: 'https://app.harness.io/'
       pathRewrite:
-        'api/proxy/harness/?': '/'
+        '/api/proxy/harness/prod/?': '/'
```
Comment (vigneswara-propelo) [inline idp-service/src/main/resources/metadata/harness-ci-cd.yaml:15]: Changes in metadata needs increment in Migration number to reflect in DB. Are we planning to HF this change?

Comment (devesh-harness) [reply idp-service/src/main/resources/metadata/harness-ci-cd.yaml:15]: ok will revert it then you can change it in your PR, HF is not confirmed yet.

**Comment by devesh-harness** [comment]
trigger uts

**Review by vigneswara-propelo — DISMISSED** [review]

**Comment by devesh-harness** [comment]
trigger uts

**Review by sathish-soundarapandian — APPROVED** [review]

**Review by vikyathharekal — APPROVED** [review]

**Comment by devesh-harness** [comment]
trigger uts

**Review by sathish-soundarapandian — APPROVED** [review]

**Review by vikyathharekal — DISMISSED** [review]

**Comment by vikyathharekal** [comment]
trigger review

**Comment by devesh-harness** [comment]
trigger ut8

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut1

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut2

**Comment by vigneswara-propelo** [comment]
trigger ut2

**Review by sarthakkasat — APPROVED** [review]

**Review by sandeepa-kv-jois — APPROVED** [review]

**Comment by devesh-harness** [comment]
trigger review

**Pull request merged**